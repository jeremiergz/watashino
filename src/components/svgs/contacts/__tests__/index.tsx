import EmailIcon from '@/components/svgs/contacts/Email';
import GitHubIcon from '@/components/svgs/contacts/GitHub';
import GitLabIcon from '@/components/svgs/contacts/GitLab';
import LinkedInIcon from '@/components/svgs/contacts/LinkedIn';
import TwitterIcon from '@/components/svgs/contacts/Twitter';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

describe('@/components/svgs/contacts Test Suite', () => {
  describe('Email Test Suite', () => {
    it('matches snapshot', () => {
      const jsx = <EmailIcon />;
      expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
    });
  });

  describe('GitHub Test Suite', () => {
    it('matches snapshot', () => {
      const jsx = <GitHubIcon />;
      expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
    });
  });

  describe('GitLab Test Suite', () => {
    it('matches snapshot', () => {
      const jsx = <GitLabIcon />;
      expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
    });
  });

  describe('LinkedIn Test Suite', () => {
    it('matches snapshot', () => {
      const jsx = <LinkedInIcon />;
      expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
    });
  });

  describe('Twitter Test Suite', () => {
    it('matches snapshot', () => {
      const jsx = <TwitterIcon />;
      expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
    });
  });
});
