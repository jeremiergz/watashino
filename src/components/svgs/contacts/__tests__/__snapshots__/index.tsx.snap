// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/components/svgs/contacts Test Suite Email Test Suite matches snapshot 1`] = `
<svg
  className=""
  height={32}
  viewBox="0 0 512 512"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    className=""
    cx="256"
    cy="256"
    r="224"
  />
  <path
    d="M257 210c-24.814 0-45 20.186-45 45s20.186 45 45 45 45-20.186 45-45-20.186-45-45-45z"
  />
  <path
    d="M255 0C114.39 0 0 114.39 0 255s114.39 257 255 257 257-116.39 257-257S395.61 0 255 0zm107 330c-20.273 0-38.152-10.161-49.017-25.596C299.23 319.971 279.354 330 257 330c-41.353 0-75-33.647-75-75s33.647-75 75-75c16.948 0 32.426 5.865 45 15.383V195c0-8.291 6.709-15 15-15s15 6.709 15 15v75c0 16.538 13.462 30 30 30s30-13.462 30-30c0-100.391-66.432-150-135-150-74.443 0-135 60.557-135 135s60.557 135 135 135c30 0 58.374-9.609 82.061-27.803 15.822-12.078 33.94 11.765 18.281 23.789C328.353 408.237 293.665 420 257 420c-90.981 0-165-74.019-165-165S166.019 90 257 90c82.897 0 165 61.135 165 180 0 33.091-26.909 60-60 60z"
  />
</svg>
`;

exports[`@/components/svgs/contacts Test Suite GitHub Test Suite matches snapshot 1`] = `
<svg
  className=""
  height={32}
  viewBox="0 0 512 512"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    className=""
    cx="256"
    cy="290"
    r="228"
  />
  <path
    d="M343.718 509.152c-2.493.922-5.001 2.036-7.567 2.497-9.829 1.799-16.12-3.404-16.178-13.705-.13-19.132-.13-38.233.174-57.365.319-22-.174-43.718-16.512-61.869 9.713-1.783 18.353-3.032 26.804-5.024 55.986-13.185 85.067-49.397 88.778-109.497 1.899-30.592-3.421-58.657-23.253-82.754-1.435-1.694-1.507-5.544-.826-8.012 5.277-19.92 3.74-39.378-2.406-58.851-1.523-4.861-4.045-6.362-8.727-6.05-16.44.936-31.502 6.853-44.78 15.95-12.801 8.815-24.718 10.733-39.925 7.165-33.487-7.745-67.192-5.485-100.607 2.453-3.783.921-8.872-.312-12.336-2.393-15.686-9.425-31.371-18.582-49.463-22.239-13.99-2.794-15.207-2.08-18.918 11.788-4.668 17.482-5.132 34.978-.363 52.563.682 2.557-.029 6.66-1.696 8.548-37.068 42.41-23.644 122.548 1.334 153.245 19.425 23.843 45.23 34.576 73.875 40.15 5.001.952 9.974 1.68 14.975 2.572-4.29 11.446-8.147 22.253-12.51 32.882-.856 2.051-3.567 3.775-5.785 4.638-28.094 11.178-52.246 2.958-68.323-23.19-8.596-13.944-19.628-24.364-36.067-27.382-.522-.104-1.059-.386-1.537-.312-4.421 1.19-8.8 2.438-13.206 3.657 2.478 3.538 4.233 8.295 7.552 10.42 13.932 9.038 22.992 21.927 28.98 37.134 9.408 23.843 27.673 34.472 51.404 36.553 11.583.966 23.282.149 35.967.149 0 14.805.522 31.707-.218 48.534-.377 9.068-9.263 12.695-19.904 8.979-28.225-9.9-54.391-24.097-76.992-44.046C13.861 394.26-17.598 303.136 9.526 196.48 36.562 90.163 107.031 24.95 212.393 4.48c145.793-28.274 283.903 80.598 298.37 232.357 11.424 119.634-57.247 231.583-167.045 272.314zM206.275 377.759c-.087.208-.145.357-.232.565.261.045.522.09.797.134.203.015.377.386-.565-.699zm-65.96 47.286c2.276 1.442 4.45 3.27 6.915 4.058.899.298 2.566-1.798 5.509-4.028-4.088-1.59-6.495-2.988-9.032-3.345-.985-.119-2.261 2.156-3.392 3.315zm20.774 3.657c.333 1.56 3.363 3.508 5.146 3.478 1.725-.03 3.407-2.378 5.132-3.716-1.566-1.472-3.131-2.899-7.944-7.343-1.044 3.092-2.769 5.589-2.334 7.58zm25.485-6.288c-2.145.52-3.957 2.557-5.915 3.954 1.146 1.487 2.29 2.869 4.132 5.203 2.537-2.379 4.262-4.029 6.002-5.634-1.45-1.293-3.146-3.79-4.219-3.523zm-63.176-8.815c1.188 2.304 3.03 4.296 5.523 7.551 1.667-2.794 3.479-4.935 3.16-5.366-1.58-2.126-3.726-3.865-5.842-5.455-.145-.12-3.088 2.764-2.841 3.27z"
  />
</svg>
`;

exports[`@/components/svgs/contacts Test Suite GitLab Test Suite matches snapshot 1`] = `
<svg
  className=""
  height={32}
  viewBox="0 0 512 512"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    className=""
    cx="256"
    cy="256"
    r="224"
  />
  <path
    d="M256 0A256 256 0 000 256a256 256 0 00256 256 256 256 0 00256-256A256 256 0 00256 0zm109.32422 94.802734c2.76229-.113738 5.58539 1.416019 6.61914 4.597657l38.81641 119.460939 19.58593 60.27344-.00195.00195a13.323818 13.323818 0 01-4.8457 14.91797L255.99414 417.20312 86.488281 294.04883c-.946726-.68776-1.754552-1.51184-2.478515-2.39453a13.309895 13.309895 0 01-2.365235-12.51953l11.382813-35.01758 8.203126-25.25 38.81445-119.455081a6.5045845 6.5045845 0 011.54883-2.490234c2.49213-2.639698 6.74639-2.677427 9.35547-.199219.2506.23947.53476.489708.75195.779297a7.6517972 7.6517972 0 011.03711 2.136719l13.09375 40.548828 25.71875 79.60938h128.88867l38.81836-119.927739c.49565-1.51754 1.39792-3.12406 2.51172-3.892579 1.05637-.73093 2.2991-1.122128 3.55469-1.173828z"
  />
</svg>
`;

exports[`@/components/svgs/contacts Test Suite LinkedIn Test Suite matches snapshot 1`] = `
<svg
  className=""
  height={32}
  viewBox="0 0 512 512"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    className=""
    cx="256"
    cy="256"
    r="224"
  />
  <path
    d="M256 0C114.637 0 0 114.637 0 256s114.637 256 256 256 256-114.637 256-256S397.363 0 256 0zm-74.39 387h-62.348V199.426h62.347zm-31.173-213.188h-.406c-20.922 0-34.453-14.402-34.453-32.402 0-18.406 13.945-32.41 35.274-32.41 21.328 0 34.453 14.004 34.859 32.41 0 18-13.531 32.402-35.274 32.402zM406.423 387h-62.34V286.652c0-25.218-9.027-42.418-31.586-42.418-17.223 0-27.48 11.602-31.988 22.801-1.649 4.008-2.051 9.61-2.051 15.215V387h-62.344s.817-169.977 0-187.574h62.344v26.558c8.285-12.78 23.11-30.96 56.188-30.96 41.02 0 71.777 26.808 71.777 84.421zm0 0"
  />
</svg>
`;

exports[`@/components/svgs/contacts Test Suite Twitter Test Suite matches snapshot 1`] = `
<svg
  className=""
  height={32}
  viewBox="0 0 512 512"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    className=""
    cx="256"
    cy="256"
    r="224"
  />
  <path
    d="M256 0C114.637 0 0 114.637 0 256s114.637 256 256 256 256-114.637 256-256S397.363 0 256 0zm116.887 199.602c.113 2.52.168 5.05.168 7.593 0 77.645-59.102 167.18-167.184 167.184h.004-.004c-33.184 0-64.062-9.727-90.066-26.395 4.597.543 9.277.813 14.015.813 27.532 0 52.867-9.39 72.98-25.152-25.722-.477-47.41-17.465-54.894-40.813a58.481 58.481 0 0011.043 1.063c5.363 0 10.559-.723 15.496-2.07-26.886-5.384-47.14-29.145-47.14-57.598 0-.266 0-.504.008-.75a58.354 58.354 0 0026.613 7.347c-15.777-10.527-26.148-28.523-26.148-48.91a58.597 58.597 0 017.957-29.535c28.976 35.555 72.28 58.937 121.117 61.394a58.708 58.708 0 01-1.528-13.398c0-32.437 26.317-58.754 58.766-58.754 16.902 0 32.168 7.145 42.89 18.567a117.855 117.855 0 0037.313-14.262c-4.394 13.715-13.707 25.222-25.84 32.5 11.887-1.422 23.215-4.574 33.742-9.254a119.412 119.412 0 01-29.308 30.43zm0 0"
  />
</svg>
`;
